hashing/te-tag-query-python/TETagQuery.py

```diff
@@ -591,6 +594,175 @@ def IDProcessor(self, idBatch, options):
             print(json.dumps(descriptor))
 
 
+# ================================================================
+class PowerSearchHandler(SubcommandHandler):
+    def __init__(self, progName, verbName):
+        super(__class__, self).__init__(progName, verbName)
+
+    def usage(self, exitCode):
+        stream = sys.stdout if exitCode == 0 else sys.stderr
+        output = """Usage: %s %s [options]
+Options:
+--limit {n} or --page-size {n}  Defines the maximum size of a page of results. The maximum is 1,000.
+--max-confidence {n}            Defines the maximum allowed confidence value for the data returned.
+--min-confidence {n}            Defines the minimum allowed confidence value for the data returned.
+--owner {x}                     Comma-separated list of AppIDs of the person who submitted the data.
+--review-status {x}             A given ReviewStatusType
+--severity {x}                  A given SeverityWType
+--share-level {x}               A given ShareLevelType
+--since {x}                     Returns descriptors collected after a timestamp
+--status {x}                    A given StatusType
+--strict-text                   When provided, the API will not do approximate
+                                matching on the value in the text field
+--tags {a,b,c}                  Comma-separated list of tags to filter results,
+                                ORed together unless --tags-are-anded is used
+--tags-are-anded                If provided, tags are ANDed together
+--text {x}                      Freeform text field with a value to search for.
+                                This can be an indicator text or a string found in other fields of the objects.
+--type or --indicator-type {x}  The type of descriptor to search for (see IndicatorTypes)
+--until {x}                     Returns descriptors collected before a timestamp
+
+Timestamp options for --since and --until are epoch seconds, various datetime
+formats including "2020-12-34T56:07:08+0900", and time-deltas like "-5minutes",
+"-3hours", "-1week".
+
+NOTE: THIS IS KNOWN TO NOT CORRECTLY HANDLE PAGINATION.  Please use
+tagged_objects with tagged_since in the TE API, which is wrapped by
+tag-to-details in this tool.
+
+See also
+https://developers.facebook.com/docs/threat-exchange/reference/apis/threat-descriptors
+""" % (
+            self.progName,
+            self.verbName,
+        )
+        stream.write(output + "\n")
+        sys.exit(exitCode)
+
+    def handle(self, args, options):
```

Comment:
(nonblocking): Out of curiosity, why hand roll parsing and help vs argparse?

Reply:
Ah. Well, I didn't use `getopt` specifically because I wanted `python te-tag-query-python --some-flags-here verb-name-here --other-flags-here` and `getopt` doesn't respect that positioning.

I didn't use `argparse` due to non-awareness. ;)

Some options I can think of:

* Leave this code as-is; no impact to be gained from changing something that already works.
* See if `argparse` respects positionality of flags before/after verb